Bugfix for opensearch statefulset

drewmarshburn opened the pull request:
### Description
Small syntax bugfix for the OpenSearch chart `StatefulSet`.

To minimally reproduce:

values.yaml
```
masterTerminationFix: true
roles:
  - master
```
 
` helm template --debug --dry-run . -f values.yaml`
```
Error: template: opensearch/templates/statefulset.yaml:410:23: executing "opensearch/templates/statefulset.yaml" at <.Values.roles.master>: can't evaluate field master in type interface {}
helm.go:84: [debug] template: opensearch/templates/statefulset.yaml:410:23: executing "opensearch/templates/statefulset.yaml" at <.Values.roles.master>: can't evaluate field master in type interface {}
```

After this fix, the `StatefulSet` can properly be templated and includes the master termination helper sidecar.

### Issues Resolved
https://github.com/opensearch-project/helm-charts/issues/226
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

peterzhuamazon commented:
Hi @drewmarshburn please follow developer guidance or #152 on contributing practices.
Could you please attach an issue to this PR, and update version/changelog, etc.?
Thanks.

DandyDeveloper reviewed (approved):
We'll need chart increments as well as the other bits mentioned by @peterzhuamazon .

This change looks good though.

drewmarshburn commented:
@peterzhuamazon @DandyDeveloper Thanks. I've added an issue, bumped the chart version, and updated the changelog. I now see that guidance in the linked issue and the contributing doc. For added clarity for others in the future, would you like me to add an adapted version of the opensearch-project `.github/PULL_REQUEST_TEMPLATE.md` (https://github.com/opensearch-project/.github/blob/main/PULL_REQUEST_TEMPLATE.md) to this repo with those additional steps?

peterzhuamazon commented:
> @peterzhuamazon @DandyDeveloper Thanks. I've added an issue, bumped the chart version, and updated the changelog. I now see that guidance in the linked issue and the contributing doc. For added convenience for others in the future, would you like me to add an adapted version of the opensearch-project `.github/PULL_REQUEST_TEMPLATE.md` (https://github.com/opensearch-project/.github/blob/main/PULL_REQUEST_TEMPLATE.md) to this repo with those additional steps?

Yeah, please go ahead and issue a PR if you can.
We welcome all the improvements to this repo from the community.
I will review this PR a bit later.

Thanks.
